lib/puppet-lint/plugins/check_unsafe_interpolations.rb

```diff
@@ -117,4 +117,13 @@ def get_exec_titles
   def interpolated?(token)
     INTERPOLATED_STRINGS.include?(token.type)
   end
+
+  # Finds the index of the next instance of `value` in `array` from the `current_index`
+  def next_instance_index(array, current_index, value)
```

Comment:
Really small nit but we should consider our parameter names.

If i read the function out of context i'm asking what is `array` - does that mean I can pass any array to it and it will do what I want?

Maybe this should be `tokens`?

Also, would calling tokens on `L104` would return all of the tokens that puppet-lint knows about? If that is the case could we give it the current token and work forward from there?

I'm not 100% on the comment above.. more or a discussion point than anything else.

Reply:
yeah i agree. Tokens would be more descriptive

Reply:
about L104... do I not need to pass the full array so that i can use `array.length` anf `array[i].value` etc??

Reply:
I can't say for sure without playing with it but I had a thought that your search would happen from position 0 and could therefore return a false value..

I wonder if we could pass a slice starting from the current index.

Reply:
tried passing a slice but this introduced some complications

ie we are iterating through indices in this new tokens array slice and so the index of the ':' found in `next_instance_index` won't be the same index as in the full tokens array

Reply:
if you don't think passing the full tokens array impacts the speed of the check too much, i'd rather stick with passing the full tokens array